subsystems/viamserver/viamserver.go

```diff
@@ -267,9 +267,9 @@ func (s *viamServer) Update(ctx context.Context, cfg *pb.DeviceSubsystemConfig,
 	return false, nil
 }
 
-func NewSubsystem(ctx context.Context, logger *zap.SugaredLogger, updateConf *pb.DeviceSubsystemConfig) (subsystems.Subsystem, error) {
+func NewSubsystem(ctx context.Context, logger logging.Logger, updateConf *pb.DeviceSubsystemConfig) (subsystems.Subsystem, error) {
 	setFastStart(updateConf)
-	return agent.NewAgentSubsystem(ctx, SubsysName, logger, &viamServer{logger: logger.Named(SubsysName)})
+	return agent.NewAgentSubsystem(ctx, SubsysName, logger, &viamServer{logger: logger.Sublogger(SubsysName)})
```

Comment:
How does the output of something with "Sublogger()" differ from the original Named() call?

Reply:
it's dotted, so it will be `viam-agent.viam-server`

`.Named()` produces a zap logger which I think has a copy not a reference; if you call this line first, then do AddAppender later, the appender doesn't show up on the zap logger from `.Named()`

Reply:
That's fine, was just curious what the difference was.